src/main/scala/ru/otus/sc/countdown/service/impl/CountdownServiceImpl.scala

```diff
@@ -0,0 +1,20 @@
+package ru.otus.sc.countdown.service.impl
+
+import ru.otus.sc.countdown.dao.CountdownDao
+import ru.otus.sc.countdown.model
+import ru.otus.sc.countdown.model.{CountdownClearRequest, CountdownDoneResponse, CountdownRequest, CountdownResponse}
+import ru.otus.sc.countdown.service.CountdownService
+
+class CountdownServiceImpl(dao: CountdownDao) extends CountdownService {
+  def countdown(request: CountdownRequest): CountdownResponse = {
+    // Can not use pattern matching here, so have to use ugly .isInstanceOf .asInstanceOf :(
+    if (request.isInstanceOf[CountdownClearRequest])
+      makeAnswer(dao.clearCountdown(request.asInstanceOf[CountdownClearRequest].initValue))
+    else makeAnswer(dao.getCountdown)
```

Comment:
`isInstanceOf` и `asInstanceOf` - важные анти-паттерны в скале. Рекомендую ознакомиться с этой старой презентацией: https://slides.yowconference.com/yowwest2014/Morris-ParametricityTypesDocumentationCodeReadability.pdf

В этом случае мы должны были использовать паттерн-матчинг:

```suggestion
    request match {
      case c: CountdownClearRequest =>
        makeAnswer(dao.clearCountdown(c.initValue))
      case _ =>
        makeAnswer(dao.getCountdown)
    }
```

Reply:
Абсолютно верно, но это сделано намеренно, в ДЗ1 мы "понятия не имеем" о патерн матчинге, поэтому здесь это так